web/src/components/game/goals/GoalManagement.tsx

```diff
@@ -128,8 +160,23 @@ export default function GoalManagement() {
                         slug={slug}
                     />
                 </Box>
+                {canModerate && (
+                    <Box sx={{ display: 'flex', justifyContent: 'flex-start', width: '100%', mt: 3 }}>
+                        <Button
+                            onClick={() => deleteDialogRef.current?.open()} // Opens delete confirmation dialog
+                            disabled={loading}
+                            color="error"
+                            variant="contained"
+                            sx={{ maxWidth: '200px' }}
+                        >
+                            {loading ? 'Deleting...' : 'Delete All Goals'}
```

Comment:
Having the delete dialog stay open, and having this loading/processing indicator feels redundant